### PR TITLE
Bug 2101732: Checking if an route exists with old syntax

### PR DESCRIPTION
**Pull request by LalatenduMohanty**
before creating the new route old as we have changed the route syntax in 8cea4f33836e0aeaf8de9cd8b1f37f569545f43f

Signed-off-by: Lalatendu Mohanty <lmohanty@redhat.com>

### CODE REVIEW

**Comment by jottofar** [comment]
Add/modify route unit tests in controllers/updateservice_controller_test.go.

File: controllers/updateservice_controller.go [hunk 1]
```diff
@@ -516,16 +516,36 @@ func validateRouteName(instance *cv1.UpdateService, name string, namespace strin
 }
 
 func (r *UpdateServiceReconciler) ensurePolicyEngineRoute(ctx context.Context, reqLogger logr.Logger, instance *cv1.UpdateService, resources *kubeResources) error {
+	oldRoute := resources.policyEngineOldRoute
 	route := resources.policyEngineRoute
+
 	// Set UpdateService instance as the owner and controller
 	if err := controllerutil.SetControllerReference(instance, route, r.Scheme); err != nil {
 		return err
 	}
```
Comment (jottofar) [inline controllers/updateservice_controller.go:525]: Shouldn't we wait to do this until we know which route, old or new, we're going to use?

File: controllers/updateservice_controller.go [hunk 2]
```diff
@@ -516,16 +516,36 @@ func validateRouteName(instance *cv1.UpdateService, name string, namespace strin
...
+		return err
+	}
+
+	if uri, _, err := routeapihelpers.IngressURI(foundOldRoute, ""); err == nil {
+		instance.Status.PolicyEngineURI = uri.String()
+	} else {
+		handleErr(reqLogger, &instance.Status, "RouteIngressFailed", err)
+	}
+
+	err = r.Client.Get(ctx, types.NamespacedName{Name: route.Name, Namespace: route.Namespace}, found)
```
Comment (jottofar) [inline controllers/updateservice_controller.go:547]: Seems we should check `oldRoutePresent` and if `true` not bother with this.

Comment (LalatenduMohanty) [reply controllers/updateservice_controller.go:547]: Right. good catch. Will fix it.

File: controllers/updateservice_controller.go [hunk 3]
```diff
@@ -516,16 +516,36 @@ func validateRouteName(instance *cv1.UpdateService, name string, namespace strin
 }
 
 func (r *UpdateServiceReconciler) ensurePolicyEngineRoute(ctx context.Context, reqLogger logr.Logger, instance *cv1.UpdateService, resources *kubeResources) error {
```
Comment (jottofar) [inline controllers/updateservice_controller.go:518]: Why not add a new method up front that returns the route, old or new, or nil. Then the rest of this method is fairly unchanged. if nil returned we operate on the new route but if a route is returned this method doesn't care whether it's old or new it just does it's thing (and my comments below are NA).

**Pull request merged**